nixos/modules/services/hardware/nvidia-container-toolkit/cdi-generate.nix

```diff
@@ -13,10 +13,10 @@
     options = mountOptions;
   };
   jqAddMountExpression = ".containerEdits.mounts[.containerEdits.mounts | length] |= . +";
-  allJqMounts = lib.concatMap
+  existingJqMounts = lib.concatMap
     (mount:
       ["${lib.getExe jq} '${jqAddMountExpression} ${builtins.toJSON (mkMount mount)}'"])
-    mounts;
+    (builtins.filter (mount: builtins.pathExists mount.hostPath) mounts);
```

Comment:
Are you sure you want `builtins.pathExists` at eval time and the eval machine, rather than at runtime at the host machine? 🙃

Reply:
That's actually a good point. Since we cannot presume that the user will provide mounts through nix paths, they could do:

hardware.nvidia-container-toolkit.mounts = [ ... { hostPath = "/some/path"; containerPath = "/some/path"; } ... ]

We cannot ensure that this will available in runtime at the nix store, so we want to perform this check on runtime, you are right. Let me check.

Reply:
Yes but more importantly the machine running `nixos-rebuild` (and doing the evaluation) is not necessarily the one on which you deploy the built system (`nixos-rebuild ... --target-host ...`), and things may change between eval time and deployment time.

TLDR: You do need to do the filtering at runtime, in a script similar to have you run jq